### PR TITLE
Additional_animated_alien_called_purple_snake

### PR DESCRIPTION
**Pull request by Carou02**
Adding an animated friendly purple snake alien

 Having a friendly animated purple snake roaming around the space station 

**Screenshots**
space_station_animated
Séquence 03

https://user-images.githubusercontent.com/87671971/158033978-74612664-93bb-4c53-a480-646bf27c9392.mp4

**Changelog**

 :cl: Carou
add: Added animated friendly purple snake, with the id: purple_snake.

### CODE REVIEW

File: Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml [hunk 1]
```diff
@@ -83,3 +83,26 @@
     heatDamageThreshold: 360
     coldDamageThreshold: -150
     currentTemperature: 310.15
+
+- type: entity
+  name: purple_snake
+  parent: SimpleSpaceMobBase
+  id: MobPurple_snake
+  description: Favorite pet corgi.
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml:91]: Don't think it's a corgi.

File: Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml [hunk 2]
```diff
@@ -83,3 +83,26 @@
...
+    sprite: Mobs/Aliens/Xenos/purple_snake.rsi
+    layers:
+    - map: ["enum.DamageStateVisualLayers.Base"]
+      state: purple_snake
+  - type: Appearance
+    visuals:
+    - type: DamageStateVisualizer
+      normal: purple_snake
+      crit: purple_snake
+      dead: purple_snake
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml:104]: Not needed if it only has 1 sprite...

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml (hunk 2 above)
Comment (Carou02) [reply Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml]: Really appreciated your feedback @metalgearsloth , did a new commit with more adapted description AND a dead state for my alien... almost forgot it.

File: Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml [hunk 3]
```diff
@@ -83,3 +83,49 @@
...
+      dead: dead_purple_snake
+  - type: Grammar
+    attributes:
+      proper: true
+      gender: male
+
+- type: entity
+  name: small_purple_snake
+  parent: SimpleSpaceMobBase
+  id: Mobsmall_purple_snake
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml:113]:
```suggestion
  id: MobSmallPurpleSnake
```

File: Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml [hunk 4]
```diff
@@ -83,3 +83,49 @@
...
+  description: A smaller version of the menacing purple snake from Kepler-283c.
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml:114]:
```suggestion
  suffix: small
  description: A smaller version of the menacing purple snake from Kepler-283c.
```

**Pull request merged**